src/Compilers/Core/Portable/InternalUtilities/Debug.cs

```diff
@@ -13,12 +13,21 @@ namespace Roslyn.Utilities
     internal static class RoslynDebug
     {
         /// <inheritdoc cref="Debug.Assert(bool)"/>
+#if NET9_0_OR_GREATER
+        [OverloadResolutionPriority(-1)]
+#endif
         [Conditional("DEBUG")]
         public static void Assert([DoesNotReturnIf(false)] bool condition) => Debug.Assert(condition);
 
         /// <inheritdoc cref="Debug.Assert(bool, string)"/>
         [Conditional("DEBUG")]
-        public static void Assert([DoesNotReturnIf(false)] bool condition, string message)
+        public static void Assert([DoesNotReturnIf(false)] bool condition,
+#if NET
+            [CallerArgumentExpression(nameof(condition))] string? message = null
```

Comment:
Why are we using the `#if NET` condition here? I get that only the core version of `Debug.Assert` has this overload but not sure if that should stop us from making `RoslynDebug.Assert` better.